extensions/machine-learning-services/src/prediction/predictService.ts

```diff
@@ -0,0 +1,202 @@
+			let result = await this._queryRunner.safeRunQuery(connection, query);
+			if (result && result.rows && result.rows.length > 0) {
+				result.rows.forEach(row => {
+					list.push({
+						databaseName: databaseName,
+						tableName: row[0].displayValue,
+						schema: row[1].displayValue
+					});
+				});
+			}
+		}
+		return list;
+	}
+
+	/**
+	 *Returns list of column names of a database
+	 * @param databaseTable table info
+	 */
+	public async getTableColumnsList(databaseTable: DatabaseTable): Promise<string[]> {
+		let connection = await this.getCurrentConnection();
+		let list: string[] = [];
+		if (connection && databaseTable.databaseName) {
+			const query = utils.withDbChange(connection.databaseName, databaseTable.databaseName, this.getTableColumnsScript(databaseTable));
+			let result = await this._queryRunner.safeRunQuery(connection, query);
+			if (result && result.rows && result.rows.length > 0) {
+				result.rows.forEach(row => {
+					list.push(row[0].displayValue);
+				});
+			}
+		}
+		return list;
+	}
+
+	private async getCurrentConnection(): Promise<azdata.connection.ConnectionProfile> {
+		return await this._apiWrapper.getCurrentConnection();
+	}
+
+	private getTableColumnsScript(databaseTable: DatabaseTable): string {
+		return `
+SELECT COLUMN_NAME,*
+FROM INFORMATION_SCHEMA.COLUMNS
+WHERE TABLE_NAME='${utils.doubleEscapeSingleQuotes(databaseTable.tableName)}'
+AND TABLE_SCHEMA='${utils.doubleEscapeSingleQuotes(databaseTable.schema)}'
+AND TABLE_CATALOG='${utils.doubleEscapeSingleQuotes(databaseTable.databaseName)}'
+		`;
+	}
+
+	private getTablesScript(databaseName: string): string {
+		return `
+SELECT TABLE_NAME,TABLE_SCHEMA
+FROM INFORMATION_SCHEMA.TABLES
+WHERE TABLE_TYPE = 'BASE TABLE' AND TABLE_CATALOG='${utils.doubleEscapeSingleQuotes(databaseName)}'
+		`;
+	}
+
+	private getPredictScriptWithModelId(
+		modelId: number,
+		columns: PredictColumn[],
+		outputColumns: PredictColumn[],
+		databaseNameTable: DatabaseTable): string {
+		return `
+DECLARE @model VARBINARY(max) = (
+	SELECT artifact_content
+	FROM ${utils.getRegisteredModelsThreePartsName(this._config)}
+	WHERE artifact_id = ${modelId}
+);
+WITH predict_input
+AS (
+	SELECT TOP 1000
+	${this.getColumnNames(columns, 'pi')}
+	FROM [${utils.doubleEscapeSingleBrackets(databaseNameTable.databaseName)}].[${databaseNameTable.schema}].[${utils.doubleEscapeSingleBrackets(databaseNameTable.tableName)}] as pi
+)
+SELECT
+${this.getInputColumnNames(columns, 'predict_input')}, ${this.getColumnNames(outputColumns, 'p')}
+FROM PREDICT(MODEL = @model, DATA = predict_input)
+WITH (
+	${this.getColumnTypes(outputColumns)}
+) AS p
+`;
+	}
+
+	private getPredictScriptWithModelBytes(
+		modelBytes: string,
+		columns: PredictColumn[],
+		outputColumns: PredictColumn[],
+		databaseNameTable: DatabaseTable): string {
+		return `
+WITH predict_input
+AS (
+	SELECT TOP 1000
```

Comment:
1000 is a picked number or could be anthing?

Reply:
I picked 1000 to be similar to "select top 1000)script. we have in OE. the ides is, this is just a sample script user will learn from to create job. we dont want to break or hang ADS if the table is too big 

---
In reply to: [389192747](https://github.com/microsoft/azuredatastudio/pull/9450#discussion_r389192747) [](ancestors = 389192747)